images/router/haproxy/conf/haproxy-config.template

```diff
@@ -33,6 +33,11 @@
 
 global
   maxconn {{env "ROUTER_MAX_CONNECTIONS" "20000"}}
+{{- if ne (env "ROUTER_THREADS") "" }}
+  nbthread {{env "ROUTER_THREADS"}}
```

Comment:
Should we be configuring [cpu-map](https://cbonte.github.io/haproxy-dconv/1.8/configuration.html#cpu-map) here to make sure threads are mapped to cpu sets (ensuring the workload is distributed)?  If we detect the number of cores and set nbthread to that there is an auto setting.

[ref](https://www.haproxy.com/blog/multithreading-in-haproxy/)

Reply:
cpu-map needs nbproc, this could be more complex , but it might be possible since we're already reading processor information.
 we could have something like this

nbproc {{env "ROUTER_NUM_CPUS"}}
cpu-map auto:/all 0-{{env "ROUTER_NUM_CPUS" - 1}} this operation is not valid on the template is just as an example
nbthread {{env "ROUTER_THREADS"}}